### DISCUSSION TITLE
Creating Github actions to make sure Xamarin apps build

### DISCUSSION THREAD

**Discussion by saamerm** [community/community#26038] [Actions]
<p>I am a Xamarin iOS &amp; Android app developer. I have a .Net solution that has an NUnitTest project to perform all my UnitTesting.</p>
<p>The target framework of the project is .NET Framework 4.5 and I’m just trying to make the Action run every time I make a commit in a Pull Request. I am using NUnit package version 2.6.4</p>
<p>I have been searching and looking far and wide to try to be able to do it, but I have wasted several days and have been extremely unsuccessful.</p>
<p>According to the Github <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners" rel="nofollow noopener">docs</a>, Github’s MacOS runners do support NUnit Test Framework v3.6.1. Can I update my package to 3.12, or will it now work on their runners anymore?</p>
<p>What are some good resources for me to sort this stuff out? Videos/Blogs/Documentation?</p>

**Comment by BrightRan** [comment]
<p>Currently, we seems have no commands or actions to upgrade NUnit in the workflow run. You need to manually download and upgrade NUnit on the machine where the runner is hosted.<br>
So if you have macOS machine or VM, you upgrade NUnit to the latest version NUnit v3.12 on the macOS machine, and install a shelf-hosted runner on the machine to run your workflow.</p>
<p>Download NUnit v3.12 from: <a href="https://github.com/nunit/nunit/releases" rel="nofollow noopener">https://github.com/nunit/nunit/releases</a><br>
NUnit Documentation: <a href="https://github.com/nunit/docs/wiki/NUnit-Documentation" rel="nofollow noopener">https://github.com/nunit/docs/wiki/NUnit-Documentation</a></p>

**Comment by saamerm** [comment, marked as answer]
<p>Alright, after months, I finally figured it out!</p>
<p>I wrote a <a href="https://medium.com/@prototypemakers/using-github-actions-with-ios-and-android-xamarin-apps-693a93b48a61" rel="nofollow noopener">blog post</a> showing how you can setup automated builds using GitHub Actions for Xamarin repos in seconds!  (<a href="https://medium.com/@prototypemakers/using-github-actions-with-ios-and-android-xamarin-apps-693a93b48a61?source=friends_link&amp;sk=cd81773f2e5a5931ae49c9362b4db795" rel="nofollow noopener">Unmetered link</a>)</p>
<p>There’s also a video in there, that shows me walking through it</p>

**Comment by PieterjanDeClippel** [comment]
<p>Paying stories don’t really have a value on this forum.</p>
